pkg/manifestcomparators/comp_conditions_must_have_proper_ssa_tags.go

```diff
@@ -0,0 +1,234 @@
+				}
+
+				return false
+			})
+
+		for _, invalidConditionProp := range invalidConditionsProperties {
+			errStr := fmt.Sprintf("crd/%v version/%v field/^.status.condition must define valid condition properties: %s", crd.Name, newVersion.Name, invalidConditionProp)
+			errsToReport = append(errsToReport, errStr)
+		}
+
+		for _, affectedField := range conditionsWithoutMapListType {
+			errStr := fmt.Sprintf("crd/%v version/%v field/%v must set x-kubernetes-list-type with value \"map\"", crd.Name, newVersion.Name, affectedField)
+			errsToReport = append(errsToReport, errStr)
+		}
+		for _, affectedField := range conditionsWithoutListMapKeysType {
+			errStr := fmt.Sprintf("crd/%v version/%v field/%v must set x-kubernetes-list-map-keys containing value \"type\"", crd.Name, newVersion.Name, affectedField)
+			errsToReport = append(errsToReport, errStr)
+		}
+	}
+
+	return ComparisonResults{
+		Name:         c.Name(),
+		WhyItMatters: c.WhyItMatters(),
+
+		Errors:   errsToReport,
+		Warnings: nil,
+		Infos:    nil,
+	}, nil
+}
+
+func (b conditionsMustHaveProperSSATags) Compare(existingCRD, newCRD *apiextensionsv1.CustomResourceDefinition) (ComparisonResults, error) {
+	return RatchetCompare(b, existingCRD, newCRD)
+}
+
+func stringSliceContainsString(a []string, s string) bool {
+	for _, v := range a {
+		if v == s {
+			return true
+		}
+	}
+	return false
+}
+
+func jsonSliceContainsString(a []apiextensionsv1.JSON, s string) bool {
+	for _, v := range a {
+		if string(v.Raw) == s {
+			return true
+		}
+	}
+	return false
+}
+
+func validateConditionProperties(properties map[string]apiextensionsv1.JSONSchemaProps) []string {
+	var errString []string
+	typeP, ok := properties["type"]
+	if !ok {
+		errString = append(errString, "type attribute is missing")
+	}
+	reason, ok := properties["reason"]
+	if !ok {
+		errString = append(errString, "reason attribute is missing")
+	}
+	status, ok := properties["status"]
+	if !ok {
+		errString = append(errString, "status attribute is missing")
+	}
+	lastTransitionTime, ok := properties["lastTransitionTime"]
+	if !ok {
+		errString = append(errString, "lastTransitioNTime attribute is missing")
+	}
+	observedGeneration, ok := properties["observedGeneration"]
+	if !ok {
+		errString = append(errString, "observedGeneration attribute is missing")
+	}
+	message, ok := properties["message"]
+	if !ok {
+		errString = append(errString, "message attribute is missing")
+	}
+
+	if len(errString) > 0 {
+		return errString
+	}
+
+	errString = append(errString, validateConditionType(typeP)...)
+	errString = append(errString, validateConditionReason(reason)...)
+	errString = append(errString, validateConditionStatus(status)...)
+	errString = append(errString, validateConditionObservedGeneration(observedGeneration)...)
+	errString = append(errString, validateConditionLastTransitionTime(lastTransitionTime)...)
+	errString = append(errString, validateConditionMessage(message)...)
```

Comment:
These are excellent!  Could you add a test that sees at least a couple of these fail?